scripts/skills/perks/perk_rf_whirling_death.nut

```diff
@@ -1,46 +1,168 @@
 this.perk_rf_whirling_death <- ::inherit("scripts/skills/skill", {
 	m = {
-		IsForceEnabled = false,
-		MinBonus = 0,
-		MaxBonus = 25
+		RequiredWeaponType = ::Const.Items.WeaponType.Flail,
+		Stacks = 0,
+		IsPerformingExtraAttack = false, // Is flipped during the extra attack to reduce its damage using onAnySkillUsed
+		DamageTotalMult = 0.5, // Multiplier for the damage dealt during the extra attack
+		ChanceToHitHeadPerReachDiff = 10 // Is used for two-handed flails
 	},
 	function create()
 	{
 		this.m.ID = "perk.rf_whirling_death";
 		this.m.Name = ::Const.Strings.PerkName.RF_WhirlingDeath;
-		this.m.Description = ::Const.Strings.PerkDescription.RF_WhirlingDeath;
+		this.m.Description = "This character\'s attacks are like a whirlwind, making it very dangerous to be near them.";
 		this.m.Icon = "ui/perks/rf_whirling_death.png";
-		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Type = ::Const.SkillType.Perk | ::Const.SkillType.StatusEffect;
 		this.m.Order = ::Const.SkillOrder.Perk;
-		this.m.IsActive = false;
-		this.m.IsStacking = false;
-		this.m.IsHidden = false;
 	}
 
-	function isEnabled()
+	function isHidden()
 	{
-		if (this.m.IsForceEnabled)
+		return this.m.Stacks == 0;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/reach.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.m.Stacks) + " [Reach|Concept.Reach]")
+		});
+		ret.push({
+			id = 20,
+			type = "text",
+			icon = "ui/tooltips/warning.png",
+			text = "Will expire upon swapping your weapon!"
+		});
+		return ret;
+	}
+
+	function onUpdate( _properties )
+	{
+		_properties.Reach += this.m.Stacks;
+	}
+
+	function onAnySkillUsed( _skill, _targetEntity, _properties )
+	{
+		if (this.m.IsPerformingExtraAttack)
+			_properties.DamageTotalMult *= this.m.DamageTotalMult;
+
+		local weapon = _skill.getItem();
+		if (weapon != null && weapon.isItemType(::Const.Items.ItemType.TwoHanded))
 		{
-			return true;
+			local reachDiff = this.getContainer().getActor().getCurrentProperties().getReach() - _targetEntity.getCurrentProperties().getReach();
+			if (reachDiff > 0)
+				_properties.HitChance[::Const.BodyPart.Head] += reachDiff * this.m.ChanceToHitHeadPerReachDiff;
 		}
+	}
 
-		if (this.getContainer().getActor().isDisarmed()) return false;
-
-		local weapon = this.getContainer().getActor().getMainhandItem();		
-		if (weapon == null || !weapon.isWeaponType(::Const.Items.WeaponType.Flail))
+	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	{
+		if (this.isSkillValid(_skill) && !this.m.IsPerformingExtraAttack)
 		{
-			return false;
+			this.m.Stacks++;
```

Comment:
Do you really not get stacks from extra attacks triggered by this perk?
The extra attacks from flail spinner seem to be counted just fine.
Also the description of this perk says "every attack" with seemingly no exceptions.

Reply:
The "extra attacks" from this perk shouldn't give additional stacks of Reach, because they are a side-effect of your main attack. It's not intended - also seemed very overpowered in my preliminary testing when you can get +4 Reach with just 2 attacks. The extra attacks from Flail Spinner aren't "part of this perk" so they can give the extra Reach.

I thought this was obvious i.e. the extra attack from this perk doesn't give the extra Reach. But I can update the tooltip to be explicit about this.